### PR TITLE
Allow switching between pan and zoom modes and x+y zoom

### PR DESCRIPTION
**Pull request by chrisadamsonmcri**
I implemented dynamic changing between zoom and pan modes by adding listeners using functions rather than in afterInit. I trigger these using buttons in the html javascript.

### CODE REVIEW

**Comment by benmccann** [comment]
@chrisadamsonmcri this PR doesn't match the existing style of the codebase. I've just finished setting up `gulp lint` in order to help you verify whether there are any style issues. Can you rebase against the latest master, run `gulp lint`, and fix any issues? Thanks!

**Comment by benmccann** [comment]
x+y drag to zoom was enabled here: https://github.com/chartjs/chartjs-plugin-zoom/commit/7f6fadd2cbfcc6f96283d8e8d8bb00c2a3ad692f

**Comment by benmccann** [comment]
This PR is a bit tough to review. It's hard to tell what code was simply moved vs changed and there are multiple functionality changes included in this PR. Perhaps it could be broken into multiple PRs?

File: chartjs-plugin-zoom.js [hunk 1]
```diff
@@ -326,12 +330,257 @@ zoomNS.zoomCumulativeDelta = 0;
...
 			scale.originalOptions = helpers.clone(scale.options);
 		});
+		
+		chartInstance.setZoomPanToSelect = function() {
+
+			var node = chartInstance.zoom.node = chartInstance.chart.ctx.canvas;
+			var options = chartInstance.options;
+			console.log(node);	
+			if (options.zoom && options.zoom.drag) {
+				console.log('Here');
```
Comment (benmccann) [inline chartjs-plugin-zoom.js]: please remove

**Comment by chrisadamsonmcri** [comment]
Apologies. I'm not working with Chart.js any more. Feel free to dump the pull request if you want.

**Comment by jledentu** [comment]
@benmccann I'll submit another PR with a different implementation

**Pull request closed without merging**